app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :likes, foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'author_id'
+  has_many :posts, foreign_key: 'author_id'
+
+  def recent_posts
+    Post.order(created_at: :desc).includes(:author).limit(3)
+  end
```

Comment:
- [ ] This `recent_posts` method is technically incorrect. Your code fetches ALL posts and then orders them, but that is not what we want. We want to fetch the recent posts belonging to the referred user, not all posts. So if the method is called on a user that has no posts it returns an empty association array. You should have this instead.

  ```
  def recent_posts
    posts.order(created_at: :desc).limit(3)
  end
  ```

Reply:
@julie-ify thank you so much for this!